### ISSUE TITLE
Should we keep `no-inline-comments` ?

### ISSUE THREAD

**Issue by ThaNarie** [mediamonks/eslint-config#67]
Consider the following example code:

```ts
const obj = {
  a: 1,
  b: true,
  c: 'string', // this one is special
}
```

The inline comment there is not allowed, instead, it expects you to move it above it:
```ts
const obj = {
  a: 1,
  b: true,
  // this one is special
  c: 'string',
}
```

---

While in general I appreciate what it tries to prevent:
* prevent long lines with comments (prettier doesn't wrap inline comments)
* prevent inconsistency of where comments are located in code, so it's easier to scan it

However, often in cases when having a more structured code (like options objects) with small annotation comments, it really gets in the way.

👍 = keep
👎 = remove
👀 = no opinion

**Comment by leroykorterink** [comment]
I don't think we should allow inline comments, I prefer the consistency of all comments on a new line. 

In what scenario are small annotation comments really necessary? If a comment is really that small, shouldn't it be part of the variable name instead?

**Comment by reindernijhoff** [comment]
No, we don't need this rule. In general, we should remove as many rules as possible.
